src/mmw/js/src/modeling/controllers.js

```diff
@@ -9,66 +9,54 @@ var $ = require('jquery'),
     models = require('./models');
 
 var ModelingController = {
-    modelPrepare: function() {
-        if (!App.map.get('areaOfInterest')) {
+    modelPrepare: function(projectId) {
+        if (!projectId && !App.map.get('areaOfInterest')) {
             router.navigate('', { trigger: true });
             return false;
         }
     },
 
-    model: function() {
-        // TODO: If you are coming to the page with a
-        // project ID in the URL, we need to load an
-        // existing  project here.
-        // i.e. var project = new models.ProjectModel({
-        //  id: XX
-        //  }).fetch
-        //  .done(init everything else);
+    model: function(projectId) {
+        var project;
 
-        // TODO: For a new project, users will eventually
-        // be able to choose which modeling package
-        // they want to use in their project. For
-        // now, the only option is TR55, so it is
-        // hard-coded here.
+        if (projectId) {
+            project = new models.ProjectModel({
+                id: projectId
+            });
+
+            project
+                .fetch()
+                .done(function(data) {
+                    App.map.set('areaOfInterest', project.get('area_of_interest'));
+                    initViews(project);
+                });
+        } else {
+            var aoi = App.map.get('areaOfInterest').geometry;
+
+            // The AoI must always be a MultiPolygon
+            if (aoi.type === 'Polygon') {
+                aoi.type = 'MultiPolygon';
+                aoi.coordinates = [aoi.coordinates];
+            }
 
-        // Init models
-        var taskModel = new models.Tr55TaskModel(),
-            currentConditions = new models.ScenarioModel({
-                name: 'Current',
-                currentConditions: true
-            }),
-            scenario1 = new models.ScenarioModel({
-                name: 'Scenario 1'
-            }),
-            scenario2 = new models.ScenarioModel({
-                name: 'Flood Scenario'
-            }),
             project = new models.ProjectModel({
                 name: 'My Project',
-                createdAt: Date.now(),
-                areaOfInterest: App.map.get('areaOfInterest'),
-                activeScenarioSlug: 'scenario-1',
-                modelPackage: new models.ModelPackageModel({
-                    name: 'TR-55',
-                    taskModel: taskModel
-                }),
+                created_at: Date.now(),
+                area_of_interest: aoi,
```

Comment:
It would be great if we could avoid mixing styles. I know that's easier said than done...